.github/workflows/java.yml

```diff
@@ -26,47 +26,76 @@ on:
             - .github/json_matrices/build-matrix.json
     workflow_dispatch:
 
+    pull_request_target:
```

Comment:
We can't use pull_request_target on our workflows as it exposes us to vulnerabilities:
> This event is given repo secrets and a full read/write GITHUB_TOKEN to boot, however there is a catch - this action only runs in the pull request's target branch, and not the pull request's branch itself.

Reply:
Will remove this logic for now anyway